temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -164,79 +165,135 @@ public WorkflowServiceStubs newClientStub() {
     return client.stubs;
   }
 
-  // Creates an in-memory service along with client stubs for use in Java code.
-  // See also createServerOnly.
+  /*
+   * Creates an in-memory service along with client stubs for use in Java code.
+   * See also createServerOnly and createInstanceOnly.
+   */
+  public TestWorkflowService() {
+    this(0);
+  }
+
+  /*
+   * Creates an in-memory service along with client stubs for use in Java code.
+   * See also createServerOnly and createInstanceOnly.
+   */
   public TestWorkflowService(boolean lockTimeSkipping) {
     this(0);
     if (lockTimeSkipping) {
       this.lockTimeSkipping("constructor");
     }
   }
 
+  /*
+   * Creates an in-memory service along with client stubs for use in Java code.
+   * See also createServerOnly and createInstanceOnly.
+   */
   public TestWorkflowService(long initialTimeMillis) {
-    store = new TestWorkflowStoreImpl(initialTimeMillis);
-    client = new Client();
-    try {
-      InProcessServerBuilder.forName(client.serverName)
-          .directExecutor()
-          .addService(this)
-          .build()
-          .start();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
-    }
-    outOfProcessServer = null;
-  }
-
-  public TestWorkflowService() {
-    this(0);
+    this(initialTimeMillis, 0, Mode.InProcess);
   }
 
   /**
    * Creates an out-of-process rather than in-process server, and does not set up a client. Useful,
    * for example, if you want to use the test service from other SDKs.
+   *
+   * @param port the port to listen on
    */
   public static TestWorkflowService createServerOnly(int port) {
+    TestWorkflowService result = new TestWorkflowService(0, port, Mode.OutOfProcess);
     log.info("Server started, listening on " + port);
-    return new TestWorkflowService(true, port);
+    return result;
   }
 
-  private TestWorkflowService(boolean isOutOfProc, int port) {
-    if (!isOutOfProc) {
-      // isOutOfProc is just here to make unambiguous constructor overloading.
-      throw new RuntimeException("Impossible.");
-    }
-    client = null;
-    store = new TestWorkflowStoreImpl(0 /* 0 means use current time */);
-    try {
-      outOfProcessServer =
-          Grpc.newServerBuilderForPort(port, InsecureServerCredentials.create())
-              .addService(this)
-              .build()
-              .start();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+  /**
+   * Creates an instance of TestWorkflowService that does not manage its own gRPC server. Useful for
+   * including in an externally managed gRPC server.
+   */
+  public static TestWorkflowService createInstanceOnly() {
+    return new TestWorkflowService(0, 0, Mode.InstanceOnly);
+  }
+
+  private enum Mode {
+    InProcess,
+    OutOfProcess,
+    InstanceOnly;
+  }
+
+  private TestWorkflowService(long initialTimeMillis, int port, Mode mode) {
```

Comment:
I'm curious what y'all think about the private uber-constructor. I didn't want to use another dummy arg (like `isOutOfProc`) to differentiate the instance-only constructor, but if you prefer that pattern, I can switch.